src/PFRICH_geo_v1.cpp

```diff
@@ -0,0 +1,1407 @@
+//  Author: C. Dilks
+//----------------------------------
+
+#include "DD4hep/DetFactoryHelper.h"
+#include "DD4hep/OpticalSurfaces.h"
+#include "DD4hep/Printout.h"
+#include "DDRec/DetectorData.h"
+#include "DDRec/Surface.h"
+
+#include <XML/Helper.h>
+#include "XML/Layering.h"
+
+#include "TVector3.h"
+
+// ROOT includes
+#include "TGDMLParse.h"
+#include "TGDMLWrite.h"
+#include "TGeoElement.h"
+#include "TGeoManager.h"
+#include "TInterpreter.h"
+#include "TUri.h"
+
+using namespace std;
+using namespace dd4hep;
+using namespace dd4hep::rec;
+
+using namespace dd4hep::detail;
+
+static Ref_t createDetector(Detector& description, xml_h e, SensitiveDetector sens) {
+
+///*--------------------------------------------------*/
+///*--------------------------------------------------*/
+///*--------------------------------------------------*/
+ //  xml::DetElement       detElem = e;
+ //  int                   detID   = detElem.id();
+ ////  xml::Component        dims    = detElem.dimensions();
+
+ //  DetElement            det(detName, detID);
+ //  sens.setType("tracker");
+ //
+
+   xml_det_t x_det    = e;
+   int       det_id   = x_det.id();
+   int       detID   = det_id;
+
+   string det_name = x_det.nameStr();
+   Material  air      = description.air();
+
+   DetElement sdet(det_name, det_id);
+
+ //  std::vector<double> rmins = {rmin2, rmin2, rmin1, rmin1, rmin2, rmin2};
+ //  std::vector<double> rmaxs = {rmax, rmax, rmax, rmax, rmax, rmax};
+ //  std::vector<double> zs    = {-length2 / 2., -length1 / 2., -length1 / 2., length1 / 2., length1 / 2., length2 / 2.};
+
+   std::vector<double> rmins = {100,  100};
+   std::vector<double> rmaxs = {1300, 1300};
+   std::vector<double> zs    = {-5000, 5000};
+
+
+ /*--------------------------------------------------*/
+ /*--------------------------------------------------*/
+   Volume     motherVol = description.pickMotherVolume(sdet);
+ //  Material  air      = description.air();
+ //
+ //  Polycone ptube(0.0, 2.0 * M_PI, rmins, rmaxs, zs);
+ //  // Tube ptube(rmin1,rmax,length2);
+ //  Volume envelope(det_name, ptube, air);
+ //
+ //  PlacedVolume env_phv = motherVol.placeVolume(envelope);
+ //  env_phv.addPhysVolID("system", det_id);
+ //  sdet.setPlacement(env_phv);
+
+   sens.setType("tracker");
+   description.invisible();
+
+   xml::DetElement       detElem = e;
+   std::string           detName = detElem.nameStr();
+   xml::Component        dims    = detElem.dimensions();
+
+ //  double vessel_radius   =  dims.attr<double>(_Unicode(rmax));;
+
+   ///*--------------------------------------------------*/
+   /// Porting in GDML volume
+
+   xml_dim_t  x_gdml(x_det.child(_U(gdmlFile)));
+   xml_dim_t  x_par(x_det.child(_U(parent)));
+
+   string     name         = x_det.nameStr();
+   string     gdml         = x_gdml.attr<string>(_U(ref));
+   string     par_nam        = x_par.nameStr();
+   DetElement det_parent   = description.detector(par_nam);
+
+   TGDMLParse parser;
```

Comment:
We should avoid importing gdml whenever possible.

Reply:
fixed